equinix/resource_fabric_connection.go

```diff
@@ -231,13 +233,15 @@ func resourceFabricConnectionUpdate(ctx context.Context, d *schema.ResourceData,
 			waitFunction = waitForConnectionProviderStatusChange
 		}
 
-		updatedConn, err = waitFunction(d.Id(), meta, ctx)
+		conn, err := waitFunction(d.Id(), meta, ctx)
 
 		if err != nil {
 			if !strings.Contains(err.Error(), "500") {
 				d.SetId("")
 			}
 			diags = append(diags, diag.Diagnostic{Severity: 2, Summary: fmt.Sprintf("connection property update completion timeout error: %v [update payload: %v] (other updates will be successful if the payload is not shown)", err, update)})
+		} else {
+			updatedConn = conn
 		}
```

Comment:
doing it this way, in case of error it will not save the last status

Reply:
Yes, this is intentional; some of the updates might fail but we want to save the status of the last successful update to the connection. We're not worried about the status of the failed update. The response won't have any details we're concerned about saving to Terraform state.